Tools/Ecore2Code/Program.cs

```diff
@@ -171,16 +174,21 @@ private void GenerateCode()
                         Console.WriteLine("Namespace mapping {0} is missing required separator =", mapping);
                         continue;
                     }
-                    Uri uri;
-                    if (!Uri.TryCreate(mapping.Substring(0, lastIdx), UriKind.Absolute, out uri))
+                    string file = mapping.Substring(lastIdx + 1);
+                    string[] uriNs = mapping.Substring(0, lastIdx).Split(new[] { '@' });
+                    
+                    if (!Uri.TryCreate(uriNs[0], UriKind.Absolute, out Uri uri))
                     {
-                        uri = new Uri(mapping.Substring(0, lastIdx), UriKind.Relative);
+                        uri = new Uri(uriNs[0], UriKind.Relative);
+                    }
+                    fileMappings.Add(uri, file);
+
+                    if (uriNs.Length == 2)
```

Comment:
I would prefer to throw an exception or display a warning in case the length is larger than 2.

Reply:
I've added the check but can't apply it to this PR since it's closed.  Do you wan't to reopen this PR to take the change?

Reply:
Yeah, I thought it would be faster to have the PR merged and have the check inserted later on, in particular because the missing check does not break anything. However, I would favor a new PR over reopening the existing one.